### PR TITLE
Feature/NR-38 add repository tests

### PR DESCRIPTION
**Pull request by lukatarman**
Related issue: #21

### CODE REVIEW

File: src/infrastructure/database/repositories/games.repository.js [hunk 1]
```diff
@@ -19,16 +19,6 @@ export class GamesRepository {
     return await this.#dbClient.getAll("games");
   }
 
-  async getXgamesWithoutPlayerHistory(amount) {
```
Comment (stas-at-ibm) [inline src/infrastructure/database/repositories/games.repository.js:22]: why is this removed? unused?

File: src/infrastructure/database/repositories/games.repository.js [hunk 2]
```diff
@@ -54,18 +44,8 @@ export class GamesRepository {
   async getXgamesCheckedMoreThanYmsAgo(amount, ms) {
     return (
       await this.#dbClient
-        .get("history_checks")
+        .get("games")
         .aggregate([
-          {
```
Comment (stas-at-ibm) [inline src/infrastructure/database/repositories/games.repository.js:59]: why is this changed? simplified?

Comment (stas-at-ibm) [inline src/infrastructure/database/repositories/games.repository.js:59]: why is this changed? simplified?

**Review by stas-at-ibm — COMMENTED** [review]
First batch.

File: src/infrastructure/database/repositories/history.checks.repository.spec.js [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+            gameId: 3,
+            checked: true,
+            found: true,
+            source: SteamApp.validDataSources.steamcharts,
+          },
+        ];
+
+        await historyChecksRepo.updateHistoryChecks(updatedHistoryChecks);
+
+        this.unresolvedResult = this.databaseClient.getAll("history_checks");
```
Comment (stas-at-ibm) [inline src/infrastructure/database/repositories/history.checks.repository.spec.js:77]: what is this? :)

File: src/infrastructure/database/repositories/steam.apps.update.timestamps.repository.spec.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
+import { initiateInMemoryDatabase } from "../in.memory.database.client.js";
+import { SteamAppsUpdateTimestampsRepository } from "./steam.apps.update.timestamps.repository.js";
+
+describe("SteamAppsUpdateTimestampsRepository", function () {
+  describe(".insertOneSteamAppsUpdateTimestamp adds a document contining the update timestamp to the collection", function () {
```
Comment (stas-at-ibm) [inline src/infrastructure/database/repositories/steam.apps.update.timestamps.repository.spec.js:5]: contining > containing

File: src/infrastructure/database/repositories/steam.apps.update.timestamps.repository.spec.js [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+      });
+    });
+  });
+
+  describe(".getLastSteamAppsUpdateTimestamp gets the last update timestamp document added to the collection.", function () {
+    describe("When the method is run", function () {
+      beforeAll(async function () {
+        this.databaseClient = await initiateInMemoryDatabase(["update_timestamps"]);
+
+        const firstDocument = { updatedOn: new Date("June 2021") };
```
Comment (stas-at-ibm) [inline src/infrastructure/database/repositories/steam.apps.update.timestamps.repository.spec.js:40]: should be three different dates, right?

**Review by stas-at-ibm — COMMENTED** [review]
Next review batch. Only the steam apps repo is left.

File: src/infrastructure/database/repositories/steam.apps.repository.spec.js [hunk 6]
```diff
@@ -0,0 +1,792 @@
+import { SteamApp } from "../../../models/steam.app.js";
+import { initiateInMemoryDatabase } from "../in.memory.database.client.js";
+import { SteamAppsRepository } from "./steam.apps.repository.js";
+
+describe("SteamAppsRepository", function () {
+  describe(".insertManySteamApps adds provided apps into the database", function () {
```
Comment (stas-at-ibm) [inline src/infrastructure/database/repositories/steam.apps.repository.spec.js:6]: Redundant test, you tested in the client that you can add multiple apps. `insertManySteamApps` is just a wrapper.
I would remove all redundant tests. Only if there is logic a test is needed.

File: src/infrastructure/database/repositories/steam.apps.repository.spec.js [hunk 7]
```diff
@@ -0,0 +1,792 @@
...
+      });
+    });
+  });
+
+  describe(".getSteamWebUntriedFilteredSteamApps retrieves a specific set of steamApps from the database.", function () {
+    describe("If three steam apps out of eigth match the filters, and the amount of 2 is provided", function () {
+      beforeAll(async function () {
+        this.databaseClient = await initiateInMemoryDatabase(["steam_apps"]);
+
+        this.databaseClient.insertMany("steam_apps", [
```
Comment (stas-at-ibm) [inline src/infrastructure/database/repositories/steam.apps.repository.spec.js:174]: Remove duplication, use a function for the mock data. You repeat the same steps in multiple tests.

**Review by stas-at-ibm — CHANGES_REQUESTED** [review]
Alright. Done with the last. Let me know if you have questions.

File: src/infrastructure/database/repositories/games.repository.js (hunk 1 above)
Comment (lukatarman) [reply src/infrastructure/database/repositories/games.repository.js:22]: Yes, unused

File: src/infrastructure/database/repositories/games.repository.js (hunk 2 above)
Comment (lukatarman) [reply src/infrastructure/database/repositories/games.repository.js:59]: Yes, simplified/refactored. These line are unnecessary.

Comment (lukatarman) [reply src/infrastructure/database/repositories/games.repository.js:59]: Yes, simplified/refactored. These line are unnecessary.

File: src/infrastructure/database/repositories/history.checks.repository.spec.js (hunk 3 above)
Comment (lukatarman) [reply src/infrastructure/database/repositories/history.checks.repository.spec.js:77]: I just wanted to check that what is returned before we await for the result is a promise. I guess it's unnecessary, will remove it.

File: src/models/history.check.js [hunk 8]
```diff
@@ -20,7 +20,7 @@ export class HistoryCheck {
     check.gameId  = game.id
     check.checked = true;
     check.found   = found;
-    check.source  = SteamApp.validDataSources.steamCharts;
+    check.source  = SteamApp.validDataSources.steamcharts;
```
[the remark this answers is not in the document]
Comment (lukatarman) [reply src/models/history.check.js:23]: Continue here

**Review by stas-at-ibm — APPROVED** [review]
Good job, lets merge it!
